### PR TITLE
Setting up Ruby on Rails App with React and Redux integrated

### PR DESCRIPTION
**Pull request by chelobotix**
In this PR I fulfilled the following tasks:
- [x] Create a new Rails app called 'hello-rails-react'.
- [x] Use the command rails new -j webpack my_new_rails_app as described in the previous lesson's tutorial.
- [x] Verify that the jsbundling-rails gem is added to your Gemfile.
- [x] Initialize your project with Git.
- [x] Install needed node dependencies as described in the previous lesson's tutorial.
- [x] Create a babel.config.js in the root folder of your project as described in the previous lesson's tutorial.
- [x] Update webpack.config.js to include the babel loader as described in the previous lesson's tutorial.
- [x] Make sure that the project has a Postgres database set up.
- [x] Create a table for storing messages and create 5 different greetings.
- [x] Create an API endpoint that selects a random greeting from the table (you will need a controller with an action and Rails route).
- [x] Create a static view that will be the root of the app.
- [x] Create the App component with react-router and render it in a static view.
- [x] Create the Greeting component that will display a greeting.
- [x] Create a store, an action, and a reducer that will connect to your API endpoint to get the random greeting.
- [x] Display the random greeting in the Greeting component.

### CODE REVIEW

File: app/javascript/application.js [hunk 1]
```diff
@@ -1,8 +1,20 @@
...
+      <BrowserRouter>
+        <Routes>
+          <Route path="/" element={<Greeting />} />
+        </Routes>
+      </BrowserRouter>
+    </Provider>
+  );
 }
 
 ReactDOM.render(<App />, document.getElementById("root"));
```
Comment (GzimAsani) [inline app/javascript/application.js]: - [x] According to the requirements there needs to be an app.js file where the router is present, and the app should be included in  this file application.js
image

**Review by GzimAsani — CHANGES_REQUESTED** [review]
### Hello, @chelobotix                   :wave:   
 Good job so far,  done There are some issues that you still need to work on to go to the next project but you are almost there!
 
 ## Highlights
 * Good job connecting back-end with front end +1: 

 ## Required Changes 

- [ ] Check the comments below :arrow_double_down: 

 ## Optional suggestions
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- N/A



Cheers and Happy coding! clapclapclap
 
 Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
 
 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/javascript/application.js (hunk 1 above)
Comment (chelobotix) [reply app/javascript/application.js]: Thank you. The app.js file has been added to component folder

**Review by Deepakdanger — APPROVED** [review]
Hi @chelobotix  

### Status: Approved ✔️

Your project is complete!  There is nothing else to say other than... it's time to merge it  🚢 
Congratulations! 🎉 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- x

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**